src/sentry/models/outbox.py

```diff
@@ -314,13 +325,32 @@ class Meta:
         "region_name", "shard_scope", "shard_identifier", "category", "object_identifier"
     )
 
+    def get_webhook_payload_from_request(self, request: HttpRequest) -> OutboxWebhookPayload:
+        return OutboxWebhookPayload(
+            method=request.method,
+            path=request.path,
+            uri=request.get_raw_uri(),
+            headers={k: v for k, v in request.headers.items()},
+            body=request.body.decode(encoding="utf-8"),
+        )
```

Comment:
This seems like it could be a classmethod on the `OutboxBase` or even a module level method.